PocketKit/Sources/PocketKit/ReaderSettings/ReaderSettingsView.swift

```diff
@@ -5,44 +5,101 @@
+                Picker(Localization.Reader.Settings.fontLabel, selection: settings.$fontFamily) {
+                    ForEach(settings.fontSet, id: \.rawValue) { family in
+                        Text(family.rawValue)
+                            .tag(family)
                     }
-
+                    .navigationBarTitleDisplayMode(.inline)
+                }
+                Stepper(
+                    Localization.Reader.Settings.fontSizeLabel,
+                    value: settings.$fontSizeAdjustment,
+                    in: settings.fontSizeAdjustmentRange,
+                    step: settings.fontSizeAdjustmentStep
+                )
+                .accessibilityIdentifier("reader-settings-font-size-stepper")
+                if settings.isPremium {
+                    Stepper(
+                        Localization.Reader.Settings.lineHeightLabel,
+                        value: settings.$lineHeightScaleFactorIndex,
+                        in: settings.settingIndexRange,
+                        step: settings.settingIndexStep
+                    )
+                    .accessibilityIdentifier("reader-settings-line-height-stepper")
                     Stepper(
-                        Localization.fontSize,
-                        value: settings.$fontSizeAdjustment,
-                        in: Constants.allowedAdjustments,
-                        step: Constants.adjustmentStep
+                        Localization.Reader.Settings.marginsLabel,
+                        value: settings.$marginsIndex,
+                        in: settings.settingIndexRange,
+                        step: settings.settingIndexStep
                     )
-                    .accessibilityIdentifier("reader-settings-stepper")
+                    .accessibilityIdentifier("reader-settings-margins-stepper")
+                    Picker(Localization.Reader.Settings.alignmentLabel, selection: settings.$alignmentOption) {
+                        ForEach(ReaderSettings.AlignmentOption.allCases, id: \.rawValue) { alignment in
+                            Text(alignment.rawValue)
+                                .tag(alignment)
+                        }
+                        .navigationBarTitleDisplayMode(.inline)
+                    }
+                } else {
+                    SettingsRowButton(
+                        title: Localization.Reader.Settings.premiumUpsellLabel,
+                        titleStyle: .settings.row.active,
+                        leadingImageAsset: .premiumIcon,
+                        leadingTintColor: Color(.ui.teal2)
+                    ) {
+                        settings.presentPremiumUpgrade()
+                    }
+                    .sheet(
+                        isPresented: $settings.isPresentingPremiumUpgrade,
+                        onDismiss: {
+                            settings.trackPremiumDismissed(dismissReason: dismissReason)
```

Comment:
Does `isPresentingPremiumUprade` get reset to `false` on dismiss? I can't seem to pick that out.

Reply:
that value is a binding, so it should get reset every time the `sheet` gets dismissed. You can verify that by capturing the value itself when it gets changed (e.g. you can add a `sink` in `ReaderSettings` and see how it changes when the sheet gets dismissed)